apps/hpc-ftsadmin/src/app/components/flow-form.tsx

```diff
@@ -0,0 +1,4094 @@
+import React, {
+  ChangeEvent,
+  useState,
+  useEffect,
+  useCallback,
+  useRef,
+} from 'react';
+import { Form, Formik, FieldArray } from 'formik';
+import tw from 'twin.macro';
+import dayjs from 'dayjs';
+import { isRight } from 'fp-ts/Either';
+import { FormikErrors } from 'formik';
+import * as t from 'io-ts';
+import { util as codecs } from '@unocha/hpc-data';
+import GppMaybeIcon from '@mui/icons-material/GppMaybe';
+import Button from '@mui/material/Button';
+import DeleteIcon from '@mui/icons-material/Delete';
+import Stack from '@mui/material/Stack';
+import ContentCopyIcon from '@mui/icons-material/ContentCopy';
+import MuiAlert from '@mui/material/Alert';
+import AlertTitle from '@mui/material/AlertTitle';
+import _, { set, values } from 'lodash';
+import useSharePath from './Hooks/SharePath';
+import { C, dialogs } from '@unocha/hpc-ui';
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  Alert,
+  IconButton,
+} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import { Environment } from '../../environments/interface';
+import { MdAdd, MdRemove, MdClose, MdCheck } from 'react-icons/md';
+import {
+  usageYears,
+  forms,
+  governingEntities,
+  fileUpload,
+  flows as flowsResponse,
+} from '@unocha/hpc-data';
+import { getEnv } from '../context';
+import { editFlowSetting, copyFlow } from '../paths';
+import { flows } from '../paths';
+import Link from '@mui/material/Link';
+import { useNavigate, unstable_usePrompt } from 'react-router-dom';
+import { id } from 'fp-ts/lib/Refinement';
+
+export type AutoCompleteSelectionType = forms.InputSelectValueType;
+
+const INPUT_SELECT_VALUE_TYPE = forms.INPUT_SELECT_VALUE_TYPE;
+
+type UniqueDataType = {
+  [key: string]: string[];
+};
+
+export interface DeleteFlowParams {
+  VersionID: number;
+  FlowID: number;
+}
+
+export interface FileAssetEntityType {
+  collection?: string;
+  createAt?: string;
+  filename?: string;
+  id?: number;
+  mimetype?: string;
+  originalname?: string;
+  path?: string;
+  size?: number;
+  updatedAt?: string;
+}
```

Comment:
To take some advantages of Typescript, you can use `Partial<T>` to avoid typing `?` in each field.

Additionally, using `type` instead of `interface` can allow you to use some of these tricks. You can checkout this article of the difference between both: [article link]( https://blog.logrocket.com/types-vs-interfaces-typescript/)

**(TL;DR: `type` tends to give you more options in Typescript except in some especific cases)**

```suggestion
export type FileAssetEntityType = Partial<{
  collection: string;
  createAt: string;
  filename: string;
  id: number;
  mimetype: string;
  originalname: string;
  path: string;
  size: number;
  updatedAt: string;
}>
```

Reply:
Also, are these names coming from somewhere or are you defining them? If it's the case that you're defining them could we use camel case everywhere?
```
...
  fileName?: string;
  mimeType?: string;
  originalName?: string;
}
```